cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs

```diff
@@ -2779,8 +2780,15 @@ readRationalUnitInterval = readRational >>= checkUnitInterval
      | q >= 0 && q <= 1 = return q
      | otherwise        = fail "Please enter a value in the range [0,1]"
 
+readFractionAsRational :: Opt.ReadM Rational
```

Comment:
@newhoggy can we implement a property test for this? We really want to know if this breaks.